server/test/unit/com/thoughtworks/go/server/service/ConfigRepoServiceIntegrationTest.java

```diff
@@ -0,0 +1,166 @@
+import com.thoughtworks.go.config.GoConfigDao;
+import com.thoughtworks.go.config.UpdateConfigCommand;
+import com.thoughtworks.go.config.materials.git.GitMaterialConfig;
+import com.thoughtworks.go.config.remote.ConfigRepoConfig;
+import com.thoughtworks.go.config.remote.ConfigReposConfig;
+import com.thoughtworks.go.domain.config.Admin;
+import com.thoughtworks.go.domain.materials.MaterialConfig;
+import com.thoughtworks.go.presentation.TriStateSelection;
+import com.thoughtworks.go.server.domain.Username;
+import com.thoughtworks.go.server.service.result.HttpLocalizedOperationResult;
+import com.thoughtworks.go.util.GoConfigFileHelper;
+import org.hamcrest.core.Is;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.io.IOException;
+
+import static java.util.Arrays.asList;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(locations = {
+        "classpath:WEB-INF/applicationContext-global.xml",
+        "classpath:WEB-INF/applicationContext-dataLocalAccess.xml",
+        "classpath:WEB-INF/applicationContext-acegi-security.xml"
+})
+public class ConfigRepoServiceIntegrationTest {
+    @Autowired
+    private GoConfigDao goConfigDao;
+    @Autowired
+    private GoConfigService goConfigService;
+    @Autowired
+    private ConfigRepoService configRepoService;
+    @Autowired
+    private EntityHashingService entityHashingService;
+
+    private String repoId, pluginId;
+    private Username user;
+    private ConfigRepoConfig configRepo;
+
+    private GoConfigFileHelper configHelper = new GoConfigFileHelper();
+
+    @Before
+    public void setUp() throws Exception {
+        user = new Username(new CaseInsensitiveString("current"));
+        UpdateConfigCommand command = goConfigService.modifyAdminPrivilegesCommand(asList(user.getUsername().toString()), new TriStateSelection(Admin.GO_SYSTEM_ADMIN, TriStateSelection.Action.add));
+        goConfigService.updateConfig(command);
+
+        this.repoId = "repo-1";
+        this.pluginId = "json-config-repo-plugin";
+        MaterialConfig repoMaterial = new GitMaterialConfig("https://foo.git", "master");
+        this.configRepo = new ConfigRepoConfig(repoMaterial, pluginId, repoId);
+
+        configHelper.usingCruiseConfigDao(goConfigDao).initializeConfigFile();
+        configHelper.onSetUp();
+
+        goConfigService.forceNotifyListeners();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        configHelper.onTearDown();
+    }
+
+    @Test
+    public void shouldFindConfigRepoWithSpecifiedId() throws Exception {
```

Comment:
@varshavaradarajan -- These are services level **integration** tests where we need to assert on the behavior of a particular endpoint. 
I agree with you that this behavior is tested at an entity level, though tomorrow if something changes under services, we won't be able to catch this broken behavior if we don't have an integration level test